design/redactors-api.md

```diff
@@ -0,0 +1,125 @@
+# Individual Redactors API
+
+Add a set of APIs to allow getting/setting redactors by name, not just as a group.
+
+## Goals
+
+- allow listing redactors by name
+- allow editing an individual redactor
+- allow creating/removing an individual redactor
+- allow storing metadata, including enabling/disabling redactors
+
+## Non Goals
+
+- preservation of ordering/comments in the redaction spec
+
+## Background
+
+Custom redactors are being moved to a more informative UI, planned to look something like this:
+![](./images/redactors-api-spec-list.png)
+API routes for individual redactors would greatly simplify implementation and move logic from frontend to backend.
+
+
+## High-Level Design
+
+Five new API routes will be added - `/api/v1/redact/list` to list redactor names, GET/POST/DELETE `/api/v1/redact/individual/{name}` to get/set/delete them by name, and GET/POST `/api/v1/redact/metadata/{name}` to get/update metadata for a redactor.
```

Comment:
I think all of the redact route should be under /api/v1/troubleshoot/redact/...

Reply:
We already have a get/set route under redact
That can of course be moved, though

Reply:
lets move them, unless redactors are going to be used outside of troubleshoot somehow?

Reply:
I'll make another proposal to move those routes then
But I'm not sure it fits with this proposal

Reply:
added that move to this proposal

Reply:
undoing the api/v1/troubleshoot move